shared/network-data-retrieval/Cargo.toml

```diff
@@ -0,0 +1,30 @@
+[package]
+name = "network-data-retrieval"
```

Comment:
Consider adding "subspace-" prefix to the crate name which will signify the DSN. We got multiple "networks" in our system: DSN, consensus Substrate chain, domain Substrate chain, etc (let alone open RPC ports). `subspace-network-data-retrieval` will point to DSN.

Reply:
I probably wouldn't use `network` here at all. But prefix is necessary indeed, the names should be such that we can publish to crates.io.

Reply:
Done, squashed, and rebased